### ISSUE TITLE
Hybrid themes: Root padding in the post editor is lost with useRootPaddingAwareAlignments

### ISSUE THREAD

**Issue by ya123123** [WordPress/gutenberg#46504]
### Description

When I create the hybrid theme and set useRootPaddingAwareAlignments to true, the root padding is lost in the post editor.

### Step-by-step reproduction instructions

1. Use WP 6.1.1 (with and without Gutenberg plugin)
2. Copy TT3 theme.json into the classic TT1 theme folder
3. Remove all CSS from TT1 style.css
4. Remove add_theme_support( 'editor-styles' ) from TT1 functions.php
5. Check post and page mobile preview

### Screenshots, screen recording, code snippet

| "useRootPaddingAwareAlignments": false  | "useRootPaddingAwareAlignments": true|
| ------------- | ------------- |
| Root padding is inserted:<br> ![padding-false](https://user-images.githubusercontent.com/114547428/207332714-417e8ca2-34cd-4f73-88da-b80ce2d02546.png)  | root padding is lost:<br>  ![padding-true](https://user-images.githubusercontent.com/114547428/207332784-98acc2c5-f6a7-4356-8dd7-ce9edcbae288.png) |

theme.json styles.spacing.padding setting:
```
		"spacing": {
			"blockGap": "1.5rem",
			"padding": {
				"top": "var(--wp--preset--spacing--40)",
				"right": "var(--wp--preset--spacing--30)",
				"bottom": "var(--wp--preset--spacing--40)",
				"left": "var(--wp--preset--spacing--30)"
			}
		},
```

### Environment info

WP 6.1.1, instawp install
TT3 and TT1 included in WP 6.1.1
Tested with and without Gutenberg plugin

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by tellthemachines** [comment]
Hi @mayuge3! I'm afraid that this is the intended behaviour. `useRootPaddingAwareAlignments` is meant for use with block templates where the whole structure of the page is composed with blocks. What it does is remove the left and right padding values from the `body` element and add them instead to the outermost block whose layout is set to `constrained`. With hybrid themes where the templates are built in PHP with static HTML elements (not blocks), there won't be a container block with `constrained` layout that we can attach the padding to, so no padding will be applied.

If you're looking to add a `theme.json` to a classic theme with only PHP templates, the best thing to do is avoid root padding and `useRootPaddingAwareAlignments` altogether, and instead add your root padding styles classic-fashion in the enqueued stylesheet. If your theme uses a combination of PHP and block templates, you may want to leverage root padding and `useRootPaddingAwareAlignments`, but in that case you'll still need some styles in the enqueued stylesheet to deal with the classic templates.

I hope this helps! Let me know if you have any further questions.

**Comment by ya123123** [comment]
Hi @tellthemachines !
I wasn't sure if it was a bug or not that the different behavior from the block theme, but now I understand that it's the intended behavior. So the outer of the post editor present in block theme, absent in hybrid, as same as the front.

Thank you for your detailed explanation.

**Comment by douglas-johnson** [comment]
If anyone else is like me and found this issue by searching "useRootPaddingAwareAligments not working in hybrid theme"; here is how I got it working for a mostly classic theme with a theme.json and some HTML template parts. I've tested this in WordPress 6.4.3 and 6.5-beta2

- Added `"useRootPaddingAwareAligments": true` to theme.json and set `styles.spacing.padding` as desired.
- Added `class="has-global-padding is-layout-constrained"` to the parent element of the post content in all relevant templates ( single, page, etc... ).
- Added the below filter on `block_editor_settings_all` to apply the constrained layout in the editor.

```php
<div class="has-global-padding is-layout-constrained">
    <?php the_content(); ?>
</div>
```

```php
/**
 * Use Root Padding in Post Editor
 *
 * @param array $settings Block editor settings.
 * @return array Updated settings.
 */
function mytheme_use_root_padding_in_post_editor( array $settings ): array {
	return array(
		...$settings,
		'postContentAttributes' => array(
			'layout' => array(
				'type' => 'constrained',
			),
		),
	);
}
add_filter( 'block_editor_settings_all', 'mytheme_use_root_padding_in_post_editor' );
```

I appreciate the advice in this issue to "avoid root padding and useRootPaddingAwareAlignments altogether" in classic themes, and that's what I've done until this point. Now I am gradually upgrading this theme for full site editing and want to opt-in to as many block theme features as I can. I thought some other people might be in the same boat.
